src/ripple/overlay/impl/PeerImp.cpp

```diff
@@ -2499,9 +2524,13 @@ PeerImp::checkValidation(
         if (app_.getOPs().recvValidation(val, std::to_string(id())) ||
             cluster())
         {
+            overlay_.checkForSquelch(
+                val->getSignerPublic(),
+                shared_from_this(),
+                protocol::mtVALIDATION);
```

Comment:
For proposal, `overlay_.checkForSquelch` is called no matter if the node decides to forward or not. For validation, it is called only if the node decides to forward. Why the difference?